signing.go

```diff
@@ -52,12 +61,20 @@ type SignerOptions struct {
 
 	// Optional map of additional keys to be inserted into the protected header
 	// of a JWS object. Some specifications which make use of JWS like to insert
-	// additional values here. All values must be JSON-serializable.
+	// additional values here. Values will be serialized by [json.Marshal] and
+	// must be valid inputs to that function.
```

Comment:
Here you include the new statement as part of the existing paragraph; in the similar comment at the top of crypter.go you split it out into its own paragraph.